### DISCUSSION TITLE
Port Forwading does not work when using Codespaces Extension locally

### DISCUSSION THREAD

**Discussion by skifli** [community/community#63002] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

When using the GitHub Codespaces extension in the VSCode app, and connecting to a Codespace, port forwading does not work. Instead it uses the localhost IP instead of the Codespace's URL, probably due to the fact that it doesn't know it is running on a Codespace. When connecting via the online app it works fine.

## VSCode app connected to Codespace
![image](https://github.com/community/community/assets/121291719/cb8fa8bd-105d-4b2e-b0b3-fe3720171917)

## Codespaces online
![image](https://github.com/community/community/assets/121291719/1c34da11-08cc-4ddb-b008-ba5d0771002a)

**Comment by mehulrajdev** [comment]
Welcome to the GitHub Community, @mehulrajdev, we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Reply by skifli** [reply]
Thanks, I just clicked the `Report a Problem` link on the GitHub Codespaces extension for VSCode so I assumed that the category would be the correct one.

**Comment by cmuto09** [comment, marked as answer]
Hey @skifli- this is by design. The browser-based version of VS Code does not have access to your localhost's IP due to OS restrictions, which is why we have to use the fully qualified domain name when you are using VS Code Online. When working on the desktop-based version we have access to `localhost`, so we point any privately forwarded ports there. This allows a stable URL across codespaces on the desktop. It is still forwarding ports to and from your codespace, we just use a different domain due to differences in permissions between the browser- and desktop-based VS Code instances.

If you require the full domain, changing the forwarded port visibility to org or public will generate the fully qualified codespace domain within VS Code desktop.

Thanks!

**Reply by skifli** [reply]
Thank you. I think I was slightly put off by the fact that the localhost URL wasn't working (which led me to assume that the URL itself was wrong), but it turns out that was a problem with my server.
